kyaml/openapi/openapi.go

```diff
@@ -26,13 +34,45 @@ type ResourceSchema struct {
 // as metadata, replicas and spec.template.spec
 func SchemaForResourceType(t yaml.TypeMeta) *ResourceSchema {
 	initSchema()
-	rs, found := schemaByResourceType[t]
+	rs, found := c.schemaByResourceType[t]
 	if !found {
 		return nil
 	}
 	return &ResourceSchema{Schema: rs}
 }
 
+// AddSchema parses s, and adds definitions from s to the global schema.
+func AddSchema(s []byte) error {
+	if c.schemaByResourceType == nil {
```

Comment:
i don't know why this `if` is here - it doesn't look at 's'
should this be in the ctor?

Reply:
moved into parse where this is used